src/services/users.serveces.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const { v4 } = require('uuid');
+
+let users = [];
+
+const getUsers = () => users;
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: v4(),
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const getUser = (id) => {
+  return users.find(user => user.id === id) || null;
+};
+
+const deleteUser = (id) => {
+  const newUsers = users.filter(user => user.id !== id);
+
+  users = newUsers;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUser(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
```

Comment:
It can be done in such way. You can merge findIndex of user with getUser func as well.

```suggestion
const updateUser = ({ id, name }) => {
  const user = getUser(id);

  const updatedUser = { ...user, name };
  
  const indexOfUser = users.findIndex(user => user.id === id);
  
  users.splice(indexOfUser, 1, updatedUser);

  return updatedUser;
};
```